c2corg_ui/static/js/gpxupload.js

```diff
@@ -68,7 +124,14 @@ app.GpxUploadController.prototype.importGpx_ = function(gpx) {
     featureProjection: 'EPSG:3857'
   });
   if (features.length > 0) {
-    this.scope_.$root.$emit('featuresUpload', features);
+    features = this.validatedFeatures_(features);
+    if (features) {
+      this.scope_.$root.$emit('featuresUpload', features);
+    } else {
+      if (goog.DEBUG) {
+        console.error('Fatal : failed to validate geometry');
```

Comment:
It can be probably removed. I was thinking that it can be easier to debug later when some other problems with images will appear...